index.go

```diff
@@ -25,11 +27,36 @@ type Indexable interface {
 }
 
 func encodeIndexKey(k indexKey) ([]byte, error) {
-	return k.encode()
+	bs, err := k.encode()
+	if err != nil {
+		return nil, err
+	}
+
+	var buf bytes.Buffer
+	gz := zlib.NewWriter(&buf)
```

Comment:
Zlib Writer creation is really expensive. You should reuse the Writer calling to `Writer.Reset()`.